libtenzir/builtins/connectors/tcp.cpp

```diff
@@ -371,16 +371,25 @@ class loader final : public plugin_loader {
 
   auto instantiate(operator_control_plane& ctrl) const
     -> std::optional<generator<chunk_ptr>> override {
-    if (not args_.tls) {
-      if (args_.tls_certfile and not args_.tls_certfile->empty()) {
-        diagnostic::warning("certificate provided, but TLS disabled")
-          .hint("add --tls to use an encrypted connection")
-          .emit(ctrl.diagnostics());
+    if (args_.tls) {
+      // Verify that the files actually exist and are readable.
+      // Ideally we'd also like to verify that the files contain valid
+      // key material, but there's no straightforward API for this in OpenSSL.
+      auto ec = std::error_code{};
+      auto* file = nullptr;
+      file = std::fopen(*args_.tls_keyfile, "r");
+      if (file) {
+        std::fclose(file);
+      } else {
+        auto error = std::strerror(errno);
+        diagnostic::error("failed to open TLS keyfile").hint(error).throw_();
       }
-      if (args_.tls_keyfile and not args_.tls_keyfile->empty()) {
-        diagnostic::warning("keyfile provided, but TLS disabled")
-          .hint("add --tls to use an encrypted connection")
-          .emit(ctrl.diagnostics());
+      auto* file = std::open(*args_.tls_certfile, "r");
+      if (file) {
+        std::fclose(file);
+      } else {
+        auto error = std::strerror(errno);
+        diagnostic::error("failed to open TLS certfile").hint(error).throw_();
```

Comment:
This within the operator's generator:

```suggestion
      diagnostic::error("failed to open TLS certfile").hint("{}", error).emit(ctrl.diagnostics());
      co_return;
```

... and `.throw_()` (without arguments) in the operator plugin's parse function.

After emitting (not throwing) an error you always want to either `co_yield` or `co_return` to let the runtime pick the error up immediately.